src/nupic/algorithms/Connections.cpp

```diff
@@ -592,7 +592,7 @@ std::ostream& operator<< (std::ostream& stream, const Connections& self)
         const auto &synData = self.dataForSynapse( syn );
         if( synData.permanence == minPermanence )
           { synapsesDead++; }
-        else if( synData.permanence == minPermanence )
+        else if( synData.permanence == maxPermanence )
```

Comment:
well, nice catch! 
Could have been #193

Reply:
It's not that bug.  This is a fix in the print diagnostics method `operator<<` which did not exist when that bug was filed.